src/Sentry.Maui/Internal/MauiNetworkStatusListener.cs

```diff
@@ -0,0 +1,68 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Maui.Internal;
+
+internal class MauiNetworkStatusListener : INetworkStatusListener
+{
+    private readonly IConnectivity _connectivity;
+    private readonly bool _hasNetworkStatusPermission;
+
+    public MauiNetworkStatusListener(IConnectivity connectivity, SentryOptions options)
+    {
+        _connectivity = connectivity;
+
+        try
+        {
+            // Checking network access will throw if we don't have permission
+            _ = connectivity.NetworkAccess;
+            _hasNetworkStatusPermission = true;
+        }
+        catch (PermissionException)
+        {
+            _hasNetworkStatusPermission = false;
+            options.DiagnosticLogger?.LogDebug("No network status permission.  Will assume device is online.");
+        }
+    }
+
+    public bool Online => !_hasNetworkStatusPermission || TreatAsOnline(_connectivity.NetworkAccess);
+
+    public async Task WaitForNetworkOnlineAsync(CancellationToken cancellationToken)
```

Comment:
Can we verify the behavior when the token provided is already cancelled? Since we call `Register` after it was cancelled, it won't fire so I wonder if it'll hang

Reply:
Good point.  Though in that case I'll just return immediately.